Checks/PWR037/README.md

```diff
@@ -1,3 +1,115 @@
 # PWR037: Potential precision loss in call to mathematical function
 
-We are working on this recommendation and it will be available soon.
+### Issue
+
+Calling a mathematical function intended for a smaller data type results in
+lower-precision operations and potential loss of information.
+
+### Actions
+
+Replace the mathematical function call with the appropriate alternative version
+that matches the input data type.
+
+### Relevance
+
+The C programming language does not support function overloading, which would
+allow multiple functions to share the same name but handle different argument
+types. To address this limitation, the standard library of C provides multiple
+versions of mathematical functions, each tailored to specific data types. For
+example:
+
+- `sqrtf()` for `float`.
+- `sqrt()` for `double`.
+- `sqrtl()` for `long double`.
+
+Using a function intended for a smaller data type, such as `sqrtf()` with a
+`double` input, causes an implicit narrowing of the input value. This implicit
+type conversion is allowed by the compiler, but results in a loss of precision
+that can propagate through chained calculations, impacting the overall accuracy
+of the program.
+
+### Code example
+
+Consider the following code, which demonstrates the loss of precision when
+using `sqrtf()` with a `double` input:
+
+```c
+// example.c
+#include <math.h>
+#include <stdio.h>
+
+__attribute__((const)) double square_root(const double value) {
+  return sqrtf(value);
+}
+
+int main() {
+  printf("Square root of 2 is: %0.15f\n", square_root(2.0));
+}
+```
+
+Compiling and running this code produces an approximation of the square root of
+2:
+
+```txt
+$ gcc --version    
+gcc (Debian 12.2.0-14) 12.2.0
+$ gcc example.c -lm -o example
+$ ./example
+Square root of 2 is: 1.41421 35381 69861
+```
+
+Note the error in the result beyond the seventh decimal place. The accurate
+approximation for the square root of 2 would be `1.41421 35623 73095`. The
+mismatch corresponds to the 7-digit precision limit of the `float` data type on
```

Comment:
Nit: FP32 may give from 6 to 9 significant decimal digits.

Reply:
If I'm not mistaken, the 6 to 9 discussion is related to what's needed for reliable back-and-forth conversion between decimal representations and binary representations. And beyond that, the generally accepted maximum accuracy in decimal places when doing floating point arithmetic would still be 7. See https://en.wikipedia.org/wiki/Single-precision_floating-point_format#IEEE_754_standard:_binary32 for more details.

In any case, if you don't mind, I believe this would be too technical for the purpose of the check entry, so I believe it would be best to leave it out :)